### ISSUE TITLE
[Bug]: Send and swap transactions are marked in a different way within the Activity tab 

### ISSUE THREAD

**Issue by sleepytanya** [MetaMask/metamask-extension#24929]
### Describe the bug

Some transactions involving sending and swapping reflected in the Activity with a single line, e.g, 'Send ETH as BUSD' while some of the send+swap transactions are marked with two lines 'Approve..." and 'Send...'. 

### Expected behavior

-

### Screenshots/Recordings

<img width="517" alt="Screenshot 2024-05-30 at 20 11 20" src="https://github.com/MetaMask/metamask-extension/assets/104780023/eaa4b7cb-318a-45fc-b301-e665654f2ac6">

<img width="474" alt="Screenshot 2024-05-30 at 19 49 15" src="https://github.com/MetaMask/metamask-extension/assets/104780023/8eb46114-a2d9-4fdc-8d4a-1a7b0cbf2099">

https://github.com/MetaMask/metamask-extension/assets/104780023/19902740-8626-496b-884c-ab9d3c3767f4

Same transaction displayed in different ways:
<img width="974" alt="Screenshot 2024-06-11 at 13 02 47" src="https://github.com/MetaMask/metamask-extension/assets/104780023/15d0d401-549e-4c11-925c-b6aaef5b5de3">

### Steps to reproduce

1. Start Send transaction
2. Select different destination token
3. Notice that send+swap transactions displayed differently within the Activity tab

### Version

dev build

### Build type

None

### Browser

Chrome, Firefox

### Operating system

MacOS

### Additional context

dev build [4c64f44](https://github.com/MetaMask/metamask-extension/pull/24914/commits/4c64f449edbfe332f60bb56db6901949af9fb5e8)

**Comment by sleepytanya** [comment]
Present on `develop `as well:

<img width="974" alt="Screenshot 2024-06-01 at 11 28 36" src="https://github.com/MetaMask/metamask-extension/assets/104780023/595e2f10-7a62-47ec-ad69-4aa349691c08">

**Comment by sleepytanya** [comment]
The issue is still present on the latest build [Slack discussion link](https://consensys.slack.com/archives/C068SFX90PN/p1717713259996549):

<img width="491" alt="Screenshot 2024-06-06 at 22 16 22" src="https://github.com/MetaMask/metamask-extension/assets/104780023/b10efa3e-dd32-4a97-a4ac-940ec310223b">

**Comment by micaelae** [comment]
2 entries for Approve + Send is expected when sending a non-native asset. The only issue i see here is that we don't display the correct amount sent when swap+sending an ERC20 token to native. For example: send 10 USDC as ETH

The value on the right should be "-10 USDC" instead of "-0 USDC"
![Screenshot 2024-06-10 at 2 08 50 PM](https://github.com/MetaMask/metamask-extension/assets/100321200/149e3390-6efb-486a-9ab9-8cbe319116cf)

**Comment by sleepytanya** [comment]
@micaelae the problem is that send+swap for non-native tokens is marked inconsistently, it can appear as 1 tx within the Activity tab or as 2 tx. 
Latest build:

<img width="560" alt="Screenshot 2024-06-10 at 17 51 47" src="https://github.com/MetaMask/metamask-extension/assets/104780023/dc48f717-084d-4a49-acd5-5335d0ca7f45">

**Comment by micaelae** [comment]
Did you have an existing approval for DAWN before the swap+send @sleepytanya ? Or did you sign an approval before the tx was confirmed?

**Comment by sleepytanya** [comment]
@micaelae No previous transactions - there are only 2 send+swap on the scrshot, fresh install.
Here you can see USDT approved two times and still showing within the Activity:

<img width="920" alt="Screenshot 2024-06-10 at 19 37 06" src="https://github.com/MetaMask/metamask-extension/assets/104780023/dfc97bb6-d15b-4996-8b17-0f165b18c476">

I can't provide more screenshots as orangefox is out of funds=).

**Comment by micaelae** [comment]
Can you send me that transaction hashes for the "Send SOL as STG" and "Send DAI as WBNB" transactions?

**Comment by sleepytanya** [comment]
The issue has been resolved, thanks to @micaelae's findings that the non-approved transactions were utilizing approvals from previous days, likely issued before we began generating the data. Consequently, these transactions were not displayed in the Activity section. This situation arose due to the testing account being accessed by multiple individuals, making it challenging to track every transaction executed.
